modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/CopyFolderMVCRenderCommand.java

```diff
@@ -31,6 +49,47 @@
 )
 public class CopyFolderMVCRenderCommand extends BaseFolderMVCRenderCommand {
 
+	@Override
+	public String render(
+			RenderRequest renderRequest, RenderResponse renderResponse)
+		throws PortletException {
+
+		try {
+			DLFolder dlFolder = _dlFolderLocalService.getDLFolder(
+				ParamUtil.getLong(renderRequest, "sourceFolderId"));
+
+			long dlFolderSize = _dlFolderLocalService.getFolderSize(
+				dlFolder.getCompanyId(), dlFolder.getGroupId(),
+				dlFolder.getTreePath());
+
+			if (_dlSizeLimitConfigurationProvider.isCopyToAllowed(
+					dlFolder.getCompanyId(), dlFolder.getGroupId(),
+					dlFolderSize)) {
+
+				return super.render(renderRequest, renderResponse);
+			}
+
+			HttpServletRequest originalHttpServletRequest =
+				_portal.getOriginalServletRequest(
+					_portal.getHttpServletRequest(renderRequest));
+
+			SessionErrors.add(
+				originalHttpServletRequest.getSession(),
+				FolderSizeLimitExceededException.class,
+				new FolderSizeLimitExceededException(
+					_dlSizeLimitConfigurationProvider.getCopyToFailInfo(
+						dlFolder.getCompanyId(), dlFolder.getGroupId(),
+						dlFolderSize, _portal.getLocale(renderRequest))));
```

Comment:
This is a bit weird. Instead of checking if the size is correct and add the error manually if not, create a private `_validate` method that does the check and throws the exception. Then catch any errors and add them to SessionErrors; this is the conventional way of handling validation issues.